model-optimizer/mo/back/preprocessing.py

```diff
@@ -0,0 +1,383 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import logging as log
+
+from mo.utils.error import Error
+from mo.utils.utils import refer_to_faq_msg
+
+import numpy as np
+
+from openvino.preprocess import PrePostProcessor        # pylint: disable=no-name-in-module,import-error
+# pylint: disable=no-name-in-module,import-error
+from openvino.runtime import Function, Layout, PartialShape, layout_helpers
+
+
+def update_mean_scale_to_dict(input_nodes: list, mean_scale_val, scale):
+    """
+    Internal function. Updates mean/scale values from array to dictionary
+    :param: input_nodes Inputs of model
+    :param: mean_scale_val Parsed 'mean_scale_val' object from command line arguments
+    :param: scale Global scale factor for all inputs from --scale command line arguments
+    """
+    if not isinstance(mean_scale_val, dict):
+        if len(mean_scale_val) != len(input_nodes):
+            raise Error('Numbers of inputs and mean/scale values do not match. ' + refer_to_faq_msg(61))
+
+        data = np.copy(mean_scale_val)
+        mean_scale_val = {}
+        for idx, node in enumerate(input_nodes):
+            names_list = list(node.get_tensor().get_names())
+            if not names_list:
+                continue
+            node_name = names_list[0]
+            mean_scale_val.update(
+                {
+                    node_name: {
+                        'mean': data[idx][0],
+                        'scale': data[idx][1]
+                    }
+                }
+            )
+
+    if scale:
+        for node in input_nodes:
+            names_list = list(node.get_tensor().get_names())
+            if not names_list:
+                continue
+            node_name = names_list[0]
+            old_val = mean_scale_val[node_name] if node_name in mean_scale_val else None
+            mean_scale_val.update(
+                {
+                    node_name: {
+                        'mean': old_val['mean'] if old_val and 'mean' in old_val else None,
+                        'scale': scale
```

Comment:
please check that previously `scale` has not been set through `scale_value` option

Reply:
Please see my reply above - if I add this check - it will not be possible to verify it in real end-to-end scenarios, as generic main.py command line parsing logic already has this check